docs/csharp/lambda-expressions.md

```diff
@@ -1,27 +1,160 @@
 ---
-title: Lambda Expressions
-description: Lambda expressions are shorthand syntax for methods and delegates. Learn when to use them.
-keywords: .NET, .NET Core
-author: dotnet-bot
-manager: wpickett
-ms.date: 06/20/2016
-ms.topic: article
-ms.prod: visual-studio-dev-14
-ms.technology: devlang-csharp
-ms.devlang: csharp
-ms.assetid: b6a0539a-8ce5-4da7-adcf-44be345a2714
+-title: Lambda Expressions
+-description: Lean to use lambda expressions, which are executable code blocks that can be passed as arguments. 
+-keywords: .NET, .NET Core, lambda epxressions, lambdas, delegates 
+-ms-author: ronpet
+-author: rpetrusha
+-manager: wpickett
+-ms.date: 11/22/2016
+-ms.topic: article 
+-ms.prod: visual-studio-dev-14
+-ms.technology: devlang-csharp
+-ms.devlang: csharp
+-ms.assetid: b6a0539a-8ce5-4da7-adcf-44be345a2714
 ---
 
-# 🔧 Lambda Expressions
-
-> **Note**
-> 
-> This topic hasn’t been written yet! 
->
-> We welcome your input to help shape the scope and approach. You can track the status and provide input on this
-> [issue](https://github.com/dotnet/docs/issues/488) at GitHub.
-> 
-> If you would like to review early drafts and outlines of this topic, please leave a note with your contact information in the issue.
->
-> Learn more about how you can contribute on [GitHub](https://github.com/dotnet/docs/blob/master/CONTRIBUTING.md).
->
+# Lambda expressions #
+
+A *lambda expression* is a block of code (an expression or a statement block) that is treated as an object. It can be passed as an argument to methods, and it can also be returned by method calls. Lambda expressions are used extensively for:
+
+- Passing the code that is to be executed to asynchronous methods, such as @System.Threading.Tasks.Task.Run(System.Action).
+
+- Writing [LINQ query expressions](linq.md).
+
+- Creating [expression trees](expression-trees-building.md).
+
+Lambda expressions are delegates. The specific delegate type of a lambda expressions depends on its parameters and return value. Lambda expressions that don't return a value correspond to a specific `Action` delegate, depending on its number of parameters. Lambda expressions that return a value correspond to a specific `Func` delegate, depending on its number of parameters. For example, a lambda expression that has two parameters but returns no value corresponds to an @System.Action%602 delegate. A lambda expression that has one parameter and returns a value corresponds to @System.Func%602 delegate.
```

Comment:
Lambda expressions are delegates, except when they are expression trees. 

Maybe "Lambda expressions are code that can be represented either as a delegate, or an expression tree that would compile to a delegate".  (I'm not sure I like that wording, improve at will.)